geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java

```diff
@@ -0,0 +1,316 @@
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import org.assertj.core.data.Offset;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+
+import org.apache.geode.internal.statistics.EnabledStatisticsClock;
+import org.apache.geode.internal.statistics.StatisticsClock;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.rules.RedisPortSupplier;
+
+public abstract class AbstractRedisInfoStatsIntegrationTest implements RedisPortSupplier {
+
+  private static final int TIMEOUT = (int) GeodeAwaitility.getTimeout().toMillis();
+  private static final String EXISTING_HASH_KEY = "Existing_Hash";
+  private static final String EXISTING_STRING_KEY = "Existing_String";
+  private static final String EXISTING_SET_KEY_1 = "Existing_Set_1";
+  private static final String EXISTING_SET_KEY_2 = "Existing_Set_2";
+
+  private Jedis jedis;
+  private static long START_TIME;
+  private static StatisticsClock statisticsClock;
+
+  private long preTestConnectionsReceived = 0;
+  private long preTestConnectedClients = 0;
+
+  private static final String COMMANDS_PROCESSED = "total_commands_processed";
+  private static final String TOTAL_CONNECTIONS_RECEIVED = "total_connections_received";
+  private static final String CONNECTED_CLIENTS = "connected_clients";
+  private static final String OPS_PERFORMED_OVER_LAST_SECOND = "instantaneous_ops_per_sec";
+  private static final String TOTAL_NETWORK_BYTES_READ = "total_net_input_bytes";
+  private static final String NETWORK_KB_READ_OVER_LAST_SECOND = "instantaneous_input_kbps";
+  private static final String UPTIME_IN_DAYS = "uptime_in_days";
+  private static final String UPTIME_IN_SECONDS = "uptime_in_seconds";
+
+  private static final AtomicInteger numInfoCalled = new AtomicInteger(0);
+
+
+  // ------------------- Setup -------------------------- //
+  @BeforeClass
+  public static void beforeClass() {
+    statisticsClock = new EnabledStatisticsClock();
+    START_TIME = statisticsClock.getTime();
+  }
+
+  @Before
+  public void before() {
+    jedis = new Jedis("localhost", getPort(), TIMEOUT);
+    numInfoCalled.set(0);
+
+    long preSetupCommandsProcessed = Long.valueOf(getInfo(jedis).get(COMMANDS_PROCESSED));
+
+    jedis.set(EXISTING_STRING_KEY, "A_Value");
+    jedis.hset(EXISTING_HASH_KEY, "Field1", "Value1");
+    jedis.sadd(EXISTING_SET_KEY_1, "m1", "m2", "m3");
+    jedis.sadd(EXISTING_SET_KEY_2, "m4", "m5", "m6");
+
+    await().atMost(Duration.ofSeconds(2))
+        .untilAsserted(() -> assertThat(
+            Long.valueOf(getInfo(jedis).get(COMMANDS_PROCESSED)) - numInfoCalled.get() + 1)
```

Comment:
these "magic numbers" still seem like they make things more complicated and confusing for anyone coming in in the future.  If it's not possible to incorporate the offset into one fo the delegate methods, can we at least assign them to variables that indicate why we need to add +1 sometimes to these calculations?